src/librustdoc/html/render.rs

```diff
@@ -1452,11 +1454,12 @@ impl<'a> SourceCollector<'a> {
         let mut href = String::new();
         clean_srcpath(&self.scx.src_root, &p, false, |component| {
             cur.push(component);
-            fs::create_dir_all(&cur).unwrap();
+            fs::create_dir_all(&cur)?;
```

Comment:
`fs::create_dir_all(&cur)?;` can be moved to after the call to `clean_srcpath` because `create_dir_all` works recursively anyway. That means that `clean_srcpath` itself doesn't need to be modified.